__tests__/hathorwallet.test.js

```diff
@@ -244,3 +245,214 @@ test('getWalletInputInfo', () => {
+      },
+      {
+        token: 'B',
+        token_data: 2,
+        value: 10,
+        decoded: { address: 'Addr1' },
+      },
+    ],
+  };
+
+  expect(await hWallet.getTxBalance(tx)).toStrictEqual({
+    'A': 1,
+    'B': -5,
+  });
+
+  expect(await hWallet.getTxBalance(tx, { includeAuthorities: true })).toStrictEqual({
+    'A': 1,
+    'B': -5,
+    'C': 0,
+  });
+
+
+  mockWalletData.mockRestore();
+  mockAddrCheck.mockRestore();
+});
+
+test('getPreProcessedData', () => {
+  const hWallet = new FakeHathorWallet();
+  hWallet.preProcessedData = {};
+
+  expect(() => {
+    hWallet.getPreProcessedData('foo')
+  }).toThrow();
+
+  hWallet.preProcessedData = {
+    foo: 123,
+  };
+
+  expect(hWallet.getPreProcessedData('foo')).toEqual(123);
+});
+
+test('setState', async () => {
+  const hWallet = new FakeHathorWallet();
+  hWallet.preProcessWalletData.mockImplementation(() => Promise.resolve());
+  hWallet.emit = () => {};
+  hWallet.state = 0;
+
+  hWallet.setState(HathorWallet.SYNCING);
+  expect(hWallet.preProcessWalletData).not.toHaveBeenCalled();
+  expect(hWallet.state).toEqual(HathorWallet.SYNCING);
+
+  hWallet.setState(HathorWallet.PROCESSING);
+  expect(hWallet.preProcessWalletData).toHaveBeenCalled();
+  expect(hWallet.state).toEqual(HathorWallet.PROCESSING);
+  hWallet.preProcessWalletData.mockClear();
+
+  hWallet.setState(HathorWallet.PROCESSING);
+  expect(hWallet.preProcessWalletData).not.toHaveBeenCalled();
+  expect(hWallet.state).toEqual(HathorWallet.PROCESSING);
+
+  hWallet.setState(HathorWallet.READY);
+  expect(hWallet.preProcessWalletData).not.toHaveBeenCalled();
+  expect(hWallet.state).toEqual(HathorWallet.READY);
+});
+
+test('preProcessWalletData', async () => {
+  const hWallet = new FakeHathorWallet();
+  hWallet.preProcessedData = {};
+
+  hWallet.getTxBalance.mockReturnValue(Promise.resolve({
+    'A': 10,
+  }));
+  hWallet.isAddressMine.mockReturnValue(true);
+
+  hWallet._getBalanceRaw.mockReturnValue({ unlocked: 5, locked: 5 });
+
+  hWallet.getFullHistory.mockReturnValue({
+    'txid1': {
+      tx_id: 'txId1',
+      timestamp: 123,
+      is_voided: false,
+      outputs: [
+        {
+          token: 'A',
+          value: 5,
+          decoded: { address: 'addr1' },
+        },
+        {
+          token: 'A',
+          value: 5,
+          decoded: { address: 'addr1', timelock: 127 },
+        }
+      ],
+      inputs: [],
+    },
+  });
+
+  hWallet.processTxQueue.mockReturnValue(Promise.resolve());
+
+  expect(() => {
+    hWallet.getPreProcessedData('tokens');
+  }).toThrow();
+  expect(() => {
+    hWallet.getPreProcessedData('historyByToken');
+  }).toThrow();
+  expect(() => {
+    hWallet.getPreProcessedData('balanceByToken');
+  }).toThrow();
+
+
+  await hWallet.preProcessWalletData();
+  expect(hWallet.processTxQueue).toHaveBeenCalled();
+
+  expect(hWallet.getPreProcessedData('tokens')).toEqual(['A']);
+  expect(hWallet.getPreProcessedData('historyByToken')).toEqual({
```

Comment:
Maybe add another check with the balance by token?

Reply:
+1

Reply:
Done